src/api/CompleteChecklistItem.ts

```diff
@@ -0,0 +1,63 @@
+import { ICheckListItem } from "api/CheckListItemApi";
+import { spWebContext } from "providers/SPWebContext";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { DateTime } from "luxon";
+import { Person, useCurrentUser } from "api/UserApi";
+
+const completeCheckListItem = (
+  item: ICheckListItem,
+  checklistItems: ICheckListItem[] | undefined,
+  currentUser: Person
+) => {
+  const [batchedSP, execute] = spWebContext.batched();
+  const batch = batchedSP.web.lists.getByTitle("CheckListItems");
+
+  // Always add the current update to the batch
+  batch.items.getById(item.Id).update({
+    CompletedById: currentUser.Id,
+    CompletedDate: DateTime.now().toISODate(),
+  });
```

Comment:
Thought for the future if we start to see cases where batches are failing so tasks aren't becoming active.  We could move this update until after the batch completes successfully.  That way the user would have to click the button to Complete again and it would attempt to activate again.  Don't think it is needed now, but just wanted to say it while I was thinking it.